From b70f2ecb71b212225f403a8f26d3d5bdca70a107 Mon Sep 17 00:00:00 2001
From: Jan Kiszka <jan.kiszka@siemens.com>
Date: Thu, 24 Nov 2016 08:27:45 +0100
Subject: [PATCH 14/32] ivshmem-net: Map shmem region as RAM

No need for special caching, simply map the shared memory region like
RAM, thus write-back. This gives us another order of magnitude in
throughput.

Signed-off-by: Jan Kiszka <jan.kiszka@siemens.com>
---
 drivers/net/ivshmem-net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ivshmem-net.c b/drivers/net/ivshmem-net.c
index b676bed2cc2e..a535cb71adde 100644
--- a/drivers/net/ivshmem-net.c
+++ b/drivers/net/ivshmem-net.c
@@ -802,7 +802,7 @@ static int ivshm_net_probe(struct pci_dev *pdev,
 	if (!devm_request_mem_region(&pdev->dev, shmaddr, shmlen, DRV_NAME))
 		return -EBUSY;
 
-	shm = devm_memremap(&pdev->dev, shmaddr, shmlen, MEMREMAP_WC);
+	shm = devm_memremap(&pdev->dev, shmaddr, shmlen, MEMREMAP_WB);
 	if (!shm)
 		return -ENOMEM;
 
-- 
2.11.0

